packages/vscode-extension/src/debugging/ProxyDebugAdapter.ts

```diff
@@ -0,0 +1,299 @@
+import fs from "fs";
+import assert from "assert";
+import { DebugSession, ErrorDestination, Event } from "@vscode/debugadapter";
+import * as vscode from "vscode";
+import { DebugProtocol } from "@vscode/debugprotocol";
+import { debug, Disposable } from "vscode";
+import { CDPProxy } from "./CDPProxy";
+import { RadonCDPProxyDelegate } from "./RadonCDPProxyDelegate";
+import { disposeAll } from "../utilities/disposables";
+import { DEBUG_CONSOLE_LOG, DEBUG_PAUSED, DEBUG_RESUMED } from "./DebugSession";
+import { CDPProfile } from "./cdp";
+import { annotateLocations, filePathForProfile } from "./cpuProfiler";
+import { SourceMapsRegistry } from "./SourceMapsRegistry";
+
+export class ProxyDebugSessionAdapterDescriptorFactory
+  implements vscode.DebugAdapterDescriptorFactory
+{
+  createDebugAdapterDescriptor(
+    session: vscode.DebugSession
+  ): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
+    return new vscode.DebugAdapterInlineImplementation(new ProxyDebugAdapter(session));
+  }
+}
+
+// strip the wildcard `*` from the sourceMapPathOverrides before passing them to the SourceMapsRegistry
+function sourceMapAliasesFromPathOverrides(
+  sourceMapPathOverrides: Record<string, string>
+): [string, string][] {
+  return Object.entries(sourceMapPathOverrides).map(([key, value]): [string, string] => {
+    if (key.endsWith("*")) {
+      key = key.slice(0, -1);
+    }
+    if (value.endsWith("*")) {
+      value = value.slice(0, -1);
+    }
+    return [key, value];
+  });
+}
+
+const CHILD_SESSION_TYPE = "radon-pwa-node";
+
+export class ProxyDebugAdapter extends DebugSession {
+  private cdpProxy: CDPProxy;
+  private disposables: Disposable[] = [];
+  private nodeDebugSession: vscode.DebugSession | null = null;
```

Comment:
we're not debugging node here right?

Reply:
We're using the `"node"` debug session from `js-debug`, so I thought the name was appropriate.